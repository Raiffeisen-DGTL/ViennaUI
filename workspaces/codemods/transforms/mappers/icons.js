export const iconsMapper = {
    ArrowRight: 'Forward',
    Exclamation: 'WarningRing',
    Down: 'SelectOpenDown',
    ArrowLeftRing: 'BackRing',
    Close: 'Close',
    Invisible: 'EyeClosed',
    ArrowUp: 'Up',
    CheckSign: 'SealCheck',
    CheckMark: 'Checkmark',
    CheckMarkRing: 'CheckmarkRing',
    Infinity: 'Infinity',
    Attention: 'WarningTr',
    Exit: 'Exit',
    ArrowsDiagonal: 'ArrowsDiag',
    Burger: 'BurgerHor',
    Favorite: 'FavEmpty',
    ArrowDownRing: 'DownRing',
    ArrowDown: 'Down',
    Buttons: 'KeyboardSettings',
    ArrowsLeftRightBlackRgb: 'ArrowsLeftRight',
    Filter: 'Filter',
    ArrowsUpDownBlackRgb: 'ArrowsTopDown',
    Download: 'Download',
    CircleCheckedDashed: 'CheckmarkPart',
    InfoRing: 'InfoRing',
    CloseRing: 'ClearRing',
    Left: 'GoLeft',
    Compass: 'Safari',
    LeftArrow: 'Back',
    CompassArrow: 'Location',
    ArrowRightRing: 'ForwardRing',
    Coordinates1: 'Coordinates',
    ArrowUpRing: 'UpRing',
    Copy: 'Copy',
    DragAndDrop: 'Drag1',
    Draggable: 'Drag',
    Enter: 'Enter',
    Pin2: 'TerminalOnMap',
    Stop: 'Cancel',
    Plus: 'Add',
    SubmitRefillLink: 'Share',
    OtherRing: 'CurrentRing',
    Mirror: 'Mirror1',
    Queue: 'InQueue',
    Up: 'SelectHide',
    PlusRing: 'AddRing',
    Trash: 'Trash',
    Lock: 'Locked',
    Other: 'MoreHor',
    Settings1: 'Settings',
    Sorting: 'Sorting',
    LogoArrow: 'Pin1',
    More: 'MoreVert',
    Refresh3: 'RefreshRepeat',
    ViewCards: 'Grid',
    Refresh2: 'RefreshRepeat',
    ViewList: 'BurgerHor',
    Refresh: 'Retry1',
    QuestionFill: 'HintFilled',
    Rewind: 'RewindLeft',
    InfoFill: 'InfoFilled',
    RewindRight2: 'CollapseRight',
    RepeatWithErrors: 'RetryWithErrors',
    RewindLeft2: 'CollapseLeft',
    Unlock: 'Unlocked',
    RewindRight: 'RewindRight',
    Question: 'HintRing',
    Right: 'GoRight',
    Save: 'Save1',
    Visible: 'EyeOpened',
    InQueue: 'ListOpen',
    InQueueBack: 'ListClose',
    Unarchive: 'Unarchive',
    Diagram: 'ChartPie',
    Schedule: 'ChartBar2',
    Schedule3: 'ChartLineGrow1',
    ScheduleGrowth: 'ChartBarGrow2',
    Scheme: 'Scheme1',
    Scheme2: 'Scheme2',
    Accounts: 'Docs',
    CheckMagnifier: 'InvoiceSearch',
    Task: 'TaskDone',
    ByReceiptNumber: 'DocNmb',
    ClipBlack: 'Attach',
    DocumentSignAndSend: 'DocSignAndSend',
    DocumentSignVisa: 'DocSignVisa',
    AccountError: 'DocError',
    Document2: 'Clipboard',
    Document3: 'Task',
    Document3Arrow: 'TaskSend',
    DocumentFavorites: 'DocFav',
    Envelope: 'Mail',
    Excel: 'Xls',
    Incoming: 'Mail',
    BetweenYourAccounts: 'AccOperations',
    NewDocument: 'DocNew',
    BriefcaseInterest: 'BriefcaseInterest',
    OfficeSupplies: 'OfficeSupplies',
    Organizations: 'Case',
    Passport: 'Passport',
    Pen: 'Pen',
    Pencil: 'Edit',
    SignChecked: 'DocSignChecked',
    SaveAsTemplate: 'TemplateSave',
    CheckSeal: 'InvoiceSeal',
    Template: 'Template',
    Check: 'Invoice2',
    CurrentAccount: 'DocVerified',
    Seal: 'SealStamp',
    SignPartlySigned: 'PaymentPartlySigned',
    ToYourAccount: 'DocRecieve',
    Check2: 'Invoice1',
    DictionaryAdd: 'AddToDic',
    SignRemove: 'DocSignRemove',
    Schedule2: 'ChartBar1',
    VerifiedTemplate: 'TemplateSaved',
    CheckInfo: 'InvoiceInfo',
    Document: 'Document',
    SignSigned: 'PaymentSigned2',
    DocumentCopy: 'Copy',
    TargetAccount: 'AccTarget',
    DocumentEdit: 'DocEdit',
    DocumentHistory: 'DocHistory',
    SavingsAccount: 'Savings',
    DocumentSeal: 'DocSeal',
    DocumentReissue: 'DocReissue',
    EditBlackRgb: 'Edit',
    DocumentSend: 'DocSend',
    CashWithdrawal: 'AtmCash1',
    Terminal4: 'Terminal',
    Transfer: 'CardToOtherBank',
    Something: 'CardRaif',
    Card: 'CardLock',
    TransferToCard: 'C2c',
    Terminal: 'Acquiring1',
    Cards: 'Cards',
    CardLock: 'CardLock',
    CardHeart: 'CardHeart',
    Open: 'CardAddNew',
    Terminal2: 'Acquiring2',
    Mastercard: 'CardMaster',
    Terminal3: 'Pos',
    Atm: 'AtmCash1',
    Atmcard: 'AtmCards1',
    Bridge: 'Bridge',
    ByCardNumber: 'CardNmb',
    Card1: 'Card',
    CardPinChange: 'CardChangePin',
    DepositFunds: 'DepositOperations',
    HandMoney: 'HandCash1',
    Contribution: 'MoneyBag',
    BundleOfMoney: 'BundleOfMoney',
    Pif: 'Pif',
    RubleTurnover: 'RoubleReturn',
    Calculator: 'Calculator',
    CurrencyEur: 'Euro',
    Euro: 'Euro',
    RubleRadial: 'RoubleRing',
    Exchange: 'Exchange2',
    ToPay: 'ToPay',
    HandMoney2: 'HandCash2',
    DebtFcpp: 'DebtFcpp',
    MoneyBag: 'MoneyBag',
    Ruble: 'Rouble',
    Dollar: 'DollarUs',
    Pay: 'Wallet2',
    RubleArrow: 'RoubleSend',
    CashTransactions: 'CashTransactions',
    PaymentOnReceipt1: 'DocRouble',
    CirculationOfMoney: 'RoubleCashflow',
    GearWheelRuble: 'RoubleGear',
    PlusMinus: 'PlusMinus',
    Credit: 'CreditCash',
    HandCard: 'CardGet',
    Portfolio: 'Case',
    CurrencyCalculator: 'CalculatorCurrency2',
    Purse2: 'Wallet1',
    Interest: 'Percent',
    Pear: 'Pear',
    Coffee: 'Coffee',
    Man2: 'Man',
    ServicesAndService: 'PersonService',
    People: 'People2',
    Furniture: 'Furniture',
    Cocktail: 'Cocktail',
    Cake: 'Cake',
    Grandfather: 'Grandfather',
    SmileNoMouth: 'FaceNoMouth',
    Candles: 'FireHouse',
    Flower: 'Flower',
    Consultation: 'Consultation',
    Tray: 'FoodService',
    LampHome: 'Lamp',
    SmileSad: 'FaceSad',
    BabyCare: 'People1',
    Stroller: 'Stroller',
    IceCream: 'IceCream',
    Champagne: 'GlassesStemware',
    Ambulance: 'Ambulance',
    Chair: 'Chair',
    Key: 'DocSign',
    TransferToAPrivatePerson: 'ToPerson',
    Girl: 'Girl',
    Window: 'Window',
    Cafe: 'Cafe',
    Man: 'Man',
    Smile: 'FaceSmile',
    Woman: 'Women',
    Grandmother: 'Grandmother',
    ManButterfly: 'ManButterfly',
    CupSport: 'Rating',
    Heart: 'Heart',
    ManRing: 'ManRing',
    SmileNormal: 'FaceNeutral',
    Candy: 'Candy',
    Cupboard: 'Cupboard',
    HeartBroken: 'HeartBroken',
    Men: 'People1',
    Cap: 'Cook',
    Fireplace: 'Fireplace',
    Monitor: 'Monitor',
    PoliceCap: 'PoliceCap1',
    Shield2: 'Shield',
    Policeman: 'ArmyCap',
    Libra: 'Libra',
    Deposit: 'Safe',
    Bell: 'Notification',
    Flasher: 'Flasher',
    PoliceCar: 'PoliceCar1',
    Fingerprint: 'Prints2',
    Court: 'Court',
    Shield: 'Insurance',
    ShieldCheck: 'ShieldCheck',
    Design: 'Wacom',
    PostServices: 'Frame',
    Theater: 'TheaterOut',
    Violin: 'Violin',
    Bust: 'Bust',
    Literature: 'Literature',
    Movie: 'Movie',
    Art: 'Music1',
    Picture: 'Picture1',
    Brush: 'Brush',
    Barrel: 'Barrel',
    Box: 'Archive',
    Canister: 'Canister',
    ConcreteMixer: 'ConcreteMixer',
    Equipment: 'Equipment',
    Excavator: 'Excavator',
    Factory: 'Factory',
    Gas: 'Gas',
    Gears: 'Gears',
    Hammers: 'Repair1',
    Lamp: 'LightBulbLamp',
    Oil: 'Oil',
    Pipes: 'Pipes',
    Screw: 'Screw',
    Tower: 'Tower',
    Tractor: 'Tractor',
    Truck: 'Truck',
    Water: 'Water',
    WateringCan: 'WateringCan',
    Wheelbarrow: 'Wheelbarrow',
    Satellite: 'Satellite3',
    Chat: 'Chat2',
    Print: 'Printer',
    Keyboard: 'Keyboard',
    BabbleQuestion: 'ChatAsk',
    SendToMail: 'MailIn',
    Support: 'Headphones1',
    Sms: 'Chat1',
    Sound: 'Sound',
    PhoneInfo: 'MobileInfo',
    QuestionLogo: 'ChatAskRaif',
    SatelliteDish: 'Satellite2',
    Entertainment: 'Console',
    Laptop: 'Laptop',
    CommunicationServices: 'MobileConnecting',
    Headphones: 'Headphones2',
    Bobbin: 'Cinema',
    Camera: 'CameraPhoto',
    Camera1: 'CameraVideo ',
    Babble: 'ChatGroup',
    BabbleLogo: 'ChatRaif',
    Mail: 'MailOut',
    Phone: 'Phone',
    PhoneHome: 'Telephone',
    Microphone: 'Microphone',
    AlcoholTobacco: 'AlcoholAndTobacco',
    Packages: 'OtherGoods',
    PriceTag: 'TagPrice3',
    Nameplate: 'NamePlate',
    BabyClothes: 'ClothesBaby',
    Basket2: 'Grocery2',
    Cashbox: 'Cashbox',
    CarDealers: 'CarDealers',
    Purchases: 'Purchases',
    RentACar: 'CarRent',
    QrCode: 'CodeQr',
    CodeRing: 'CodeBarRing',
    Code: 'CodeBar 2',
    Kids: 'Kids',
    HomeAndAppliances: 'HomeEquipment',
    Footwear: 'Footwear2',
    Beauty: 'Mirror1',
    Clothes: 'Clothes2',
    ClothesShoesAccessories: 'Clothes1',
    Cosmetics: 'Cosmetics2',
    GasStation: 'GasStation',
    HealthAndBeauty: 'Apple',
    Offer: 'Offer',
    PriceTagRuble: 'TagPrice1',
    Products: 'Basket',
    Treatment: 'Treatment2',
    Pharmacies: 'Pharmacy',
    Moon: 'Moon',
    Rain: 'Rain',
    PetSupplies: 'Pet',
    Garden: 'Garden',
    Dolphin: 'Dolphin',
    Cloud: 'Bear',
    Snow: 'Snow',
    Tree: 'Tree',
    Fish: 'Fish',
    Leaf: 'Leaf',
    Sun: 'Sun',
    SunCloud: 'SunCloud',
    Bear: 'Bear1',
    Umbrella2: 'UmbrellaClosed',
    Cow: 'Bull',
    Animal: 'Cat',
    Umbrella: 'UmbrellaOpened',
    Whale: 'Whale',
    Ufo: 'Ufo',
    Electronics: 'Monitor',
    DutyFree: 'CreditConsumer',
    Timer: 'Timer',
    Balloon: 'Balloon',
    Backpack2: 'Suitcase',
    Cup: 'Cup',
    Tent: 'Tent',
    TennisBall: 'TennisBall',
    Skates: 'Skates',
    Fishing: 'Fishing',
    Chess: 'Chess',
    Archer: 'Archer',
    PalmBlack: 'Palm',
    Backpack: 'Backpack',
    Barbell: 'Barbell',
    Bowling: 'Bowling',
    BoxSport: 'Boxing',
    ChaiseLounge: 'Sunshade',
    Darts: 'Darts',
    Drone: 'Drone',
    Flag: 'Flag',
    Goals: 'Goals',
    Golf: 'Golf',
    Leisure: 'Tennis',
    LeisureAndTravel: 'Paris',
    Map: 'MapSearch',
    Medal: 'Medal',
    Mountains: 'Mountains',
    Parachute: 'Parachute',
    PingPong: 'PingPong',
    Robot: 'Robot',
    Skiing: 'Skiing',
    Sport: 'Basketball',
    Tourism: 'Navigation',
    Toys: 'Bear',
    TrainingApparatus: 'Gym',
    Volleyball: 'Volleyball',
    Wave2: 'Boat',
    Wave: 'Wave',
    BankHome: 'BankAndHome',
    HouseWap: 'HouseWap',
    BankIntroduction: 'BranchIn',
    Building: 'Company',
    Bank: 'Branch',
    Castle: 'Castle',
    Crane: 'Crane1',
    Crane2: 'Crane2',
    Fence: 'Fence',
    Heating: 'Heating2',
    Home: 'Home',
    Hotels: 'Hotels',
    Houses: 'City',
    Houses2: 'Houses',
    Repairs: 'Repair2',
    ToBudget: 'ToBudget',
    GeneralEducation: 'Book',
    ArtAndLiterature: 'ArtAndLiterature',
    Binoculars: 'Binoculars1',
    Binoculars2: 'Binoculars2',
    BooksShelf: 'BooksShelf',
    Courses: 'Courses',
    Education: 'Education',
    Glasses: 'GlassesEye',
    Globe: 'Globe2',
    Magnifier: 'Search',
    Microscope: 'Microscope',
    ProfessionalEducation: 'Repair1',
    WorldCopy: 'World',
    Metro: 'Metro',
    Gbdd: 'Fines',
    Cruises: 'Ship',
    Auto: 'AutotransportFront',
    Rocket: 'Rocket',
    Taxi2: 'Taxi',
    Taxi: 'Taxi',
    Speedometer: 'Speedometer',
    MetroTrain: 'Railway2',
    AutoParts: 'CarParts',
    Moto: 'Moto1',
    TrolleyBus: 'TrolleyBus',
    AirTransport: 'Airplane',
    AutoService: 'Wrench',
    Bicycle: 'Moto2',
    Helicopter: 'Helicopter',
    MotorTransport: 'Bus',
    Railway: 'Railway',
    Calendar: 'Calendar',
    Clock: 'History',
    OperationsHistory: 'Retry2',
    Expectation: 'InProgress',
    Accessories: 'Accessories',
    AdEvent: 'CalendarAdd',
    Telegram: 'Telegram',
    Whatsapp: 'Whatsapp',
    Viber: 'PhoneActive',
    Clip: 'Attach',
};
