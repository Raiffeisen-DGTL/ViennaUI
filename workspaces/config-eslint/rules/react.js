module.exports = {
    plugins: ['react'],
    parserOptions: {
        ecmaFeatures: {
            jsx: true,
        },
    },
    settings: {
        react: {
            version: 'latest',
        },
    },
    rules: {
        'jsx-quotes': [2, 'prefer-single'],

        'react/boolean-prop-naming': 0,
        'react/button-has-type': 0,
        'react/default-props-match-prop-types': 0,
        'react/destructuring-assignment': 0,
        'react/display-name': 0,
        'react/forbid-component-props': [1, { forbid: ['style'] }],
        'react/forbid-dom-props': 0,
        'react/forbid-elements': 0,
        'react/forbid-prop-types': 0,
        'react/forbid-foreign-prop-types': 0,
        'react/no-access-state-in-setstate': 0,
        'react/no-array-index-key': 1,
        'react/no-children-prop': 2,
        'react/no-danger': 2,
        'react/no-danger-with-children': 2,
        'react/no-deprecated': 2,
        'react/no-did-mount-set-state': 0,
        'react/no-did-update-set-state': 0,
        'react/no-direct-mutation-state': 2,
        'react/no-find-dom-node': 1,
        'react/no-is-mounted': 2,
        'react/no-multi-comp': [2, { ignoreStateless: true }],
        'react/no-redundant-should-component-update': 0,
        'react/no-render-return-value': 2,
        'react/no-set-state': 0,
        'react/no-typos': 2,
        'react/no-string-refs': 2,
        'react/no-unescaped-entities': 0,
        'react/no-unknown-property': 2,
        'react/no-unused-prop-types': 2,
        'react/no-unused-state': 2,
        'react/no-will-update-set-state': 2,
        'react/prefer-es6-class': 2,
        'react/prefer-stateless-function': [2, { ignorePureComponents: true }],
        'react/prop-types': 2,
        'react/react-in-jsx-scope': 2,
        'react/require-default-props': 0,
        'react/require-optimization': 0,
        'react/require-render-return': 2,
        'react/self-closing-comp': [2, { html: false }],
        'react/sort-comp': [
            2,
            {
                order: [
                    'variables',
                    'static-variables',
                    'constructor',
                    'static-methods',
                    'lifecycle',
                    'rendering',
                    '/^handle.+$/',
                    'instance-variables',
                    'instance-methods',
                    'everything-else',
                ],
                groups: {
                    variables: ['/animation|ID|timeout|interval|node|delay/gi', '/^el.+$/'],
                    lifecycle: [
                        'contextTypes',
                        'childContextTypes',
                        'defaultProps',
                        'propTypes',
                        'state',
                        'constructor',
                        'getChildContext',
                        'getDerivedStateFromProps',
                        'componentWillMount',
                        'UNSAFE_componentWillMount',
                        'componentDidMount',
                        'componentWillReceiveProps',
                        'UNSAFE_componentWillReceiveProps',
                        'shouldComponentUpdate',
                        'componentWillUpdate',
                        'UNSAFE_componentWillUpdate',
                        'getSnapshotBeforeUpdate',
                        'componentDidUpdate',
                        'componentDidCatch',
                        'componentWillUnmount',
                    ],
                    rendering: ['render', '/^render.+$/'],
                },
            },
        ],
        'react/sort-prop-types': 0,
        'react/style-prop-object': 1,
        'react/void-dom-elements-no-children': 2,

        'react/jsx-boolean-value': [1, 'never'],
        'react/jsx-closing-bracket-location': [1, { selfClosing: 'line-aligned', nonEmpty: 'after-props' }],
        'react/jsx-closing-tag-location': 2,
        'react/jsx-curly-spacing': 2,
        'react/jsx-equals-spacing': 2,
        'react/jsx-filename-extension': 0,
        'react/jsx-first-prop-new-line': [2, 'multiline'],
        'react/jsx-handler-names': 2,
        'react/jsx-indent': 'off',
        'react/jsx-indent-props': [2, 4],
        'react/jsx-key': 2,
        'react/jsx-max-props-per-line': [1, { when: 'multiline' }],
        'react/jsx-no-bind': [1, { ignoreRefs: true }],
        'react/jsx-no-comment-textnodes': 2,
        'react/jsx-no-duplicate-props': [2, { ignoreCase: true }],
        'react/jsx-no-literals': 0,
        'react/jsx-no-target-blank': 2,
        'react/jsx-no-undef': 2,
        'react/jsx-one-expression-per-line': 0,
        'react/jsx-curly-brace-presence': [0, { props: 'always', children: 'never' }],
        'react/jsx-pascal-case': 2,
        'react/jsx-sort-props': [
            1,
            {
                callbacksLast: true,
                noSortAlphabetically: true,
                reservedFirst: ['key'],
            },
        ],
        'react/jsx-tag-spacing': [
            2,
            {
                closingSlash: 'never',
                beforeSelfClosing: 'always',
                afterOpening: 'never',
            },
        ],
        'react/jsx-uses-react': 2,
        'react/jsx-uses-vars': 2,
        'react/jsx-wrap-multilines': 2,
    },
};
