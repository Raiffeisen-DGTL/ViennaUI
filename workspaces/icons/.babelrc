{
  "presets": ["@babel/env", "@babel/typescript", "@babel/react"],
  "env": {
    "esm": {
      "presets": [["@babel/env", { "modules": false }]]
    }
  }
}
