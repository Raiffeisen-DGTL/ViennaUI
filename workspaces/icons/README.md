# Библиотека иконок

###### Raiffeisenbank Design System

Коллекция иконок для использования в интерфейсах. Иконки представлены в виде React компонентов и оригинальных векторных (`*.svg`) изображений. Изображения напрямую можно использовать из пакета.
