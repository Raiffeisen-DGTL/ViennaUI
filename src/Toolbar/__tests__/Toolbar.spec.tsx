import React from 'react';
import { Screw } from '@fcc/icons';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Toolbar } from '../index';

describe('Toolbar', () => {
    it('should pass any props', () => {
        const props = {
            className: 'className',
            id: 'id',
            title: 'title',
        };

        const component = (
            <Toolbar {...props}>
                <Toolbar.Operation label={'Option 1'} />
                <Toolbar.Operation label={'Option 2'} />
                <Toolbar.Operation label={'Option 3'} />
                <Toolbar.Operation label={'Option 4'} />
            </Toolbar>
        );

        const snap = snapshot.render(component);
        expect(snap).toMatchSnapshot();

        const wrapper = snapshot.shallow(component);
        for (const [key, value] of Object.entries(props)) {
            expect(wrapper.props()[key]).toEqual(value);
        }
    });

    xit('should onClick on operation when click', () => {
        const fn = jest.fn();
        render(<Toolbar.Operation id={'id'} name={'name'} icon={<Screw />} label={'Action 5'} onClick={fn} />);

        const operation = screen.getByRole('button');
        userEvent.click(operation);

        expect(fn).toHaveBeenCalledWith(expect.any(Object), { id: 'id', name: 'name' });
    });

    xit('should check the functionality of hover to the operation', () => {
        render(
            <Toolbar.Operation label={'Action 3'} data-marker='action'>
                <Toolbar.Operation label={'Option 1'} data-marker='option' />
                <Toolbar.Operation label={'Option 2'} />
                <Toolbar.Operation label={'Option 3'} />
            </Toolbar.Operation>
        );

        const action = screen.getByTestId('action');
        const option = screen.queryByTestId('option');
        expect(option).not.toBeInTheDocument();

        userEvent.hover(action);
        expect(screen.queryByTestId('option')).toBeInTheDocument();

        userEvent.unhover(action);
        expect(screen.queryByTestId('option')).not.toBeInTheDocument();
    });

    xit('should check the functionality of hover for nested operation', () => {
        render(
            <Toolbar.Operation label={'Action 3'} data-marker='action'>
                <Toolbar.Operation label={'Option 1'} data-marker='option'>
                    <Toolbar.Operation label={'Option A'} data-marker='option-a' />
                    <Toolbar.Operation label={'Option B'} />
                </Toolbar.Operation>
                <Toolbar.Operation label={'Option 2'} />
                <Toolbar.Operation label={'Option 3'} />
            </Toolbar.Operation>
        );

        const action = screen.getByTestId('action');

        userEvent.hover(action);
        const option = screen.getByTestId('option');

        userEvent.hover(option);
        expect(screen.queryByTestId('option-a')).toBeInTheDocument();

        userEvent.unhover(option);
        userEvent.unhover(action);
        expect(screen.queryByTestId('option')).not.toBeInTheDocument();
        expect(screen.queryByTestId('option-a')).not.toBeInTheDocument();
    });

    it('should check that the functionality of click works with "loading" and "disabled" props', () => {
        const fn1 = jest.fn();
        const fn2 = jest.fn();
        const fn3 = jest.fn();

        render(
            <>
                <Toolbar.Operation
                    label={'Option 1'}
                    onClick={fn1}
                    data-testid='option-1'
                    id={'id1'}
                    name={'name1'}
                    loading
                />
                <Toolbar.Operation
                    label={'Option 2'}
                    onClick={fn2}
                    data-testid='option-2'
                    id={'id2'}
                    name={'name3'}
                    disabled
                />
                <Toolbar.Operation label={'Option 3'} onClick={fn3} data-testid='option-3' id={'id3'} name={'name3'} />
            </>
        );

        const option1 = screen.getByTestId('option-1');
        const option2 = screen.getByTestId('option-2');
        const option3 = screen.getByTestId('option-3');

        fireEvent.click(option1);
        expect(fn1).not.toHaveBeenCalled();

        fireEvent.click(option2);
        expect(fn2).not.toHaveBeenCalled();

        fireEvent.click(option3);
        expect(fn3).toHaveBeenCalled();
    });
});
