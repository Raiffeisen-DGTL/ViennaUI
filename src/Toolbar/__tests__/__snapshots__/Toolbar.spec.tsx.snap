// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar should pass any props 1`] = `
.c4 {
  margin-left: auto;
  margin-right: 0em;
}

.c0 {
  outline: none;
  box-sizing: border-box;
  width: 100%;
  display: flex;
  flex-direction: row;
  align-items: center;
  height: 62px;
  padding-left: 24px;
  padding-right: 24px;
}

.c1 {
  padding-left: 32px;
  cursor: pointer;
  display: flex;
  align-items: center;
  position: relative;
  min-width: auto;
}

.c1:first-child {
  padding-left: 0px;
}

.c2 {
  display: inline-flex;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  white-space: nowrap;
  user-select: none;
  appearance: none;
  cursor: pointer;
  outline: 0;
  position: relative;
  border-radius: 8px;
  border-color: transparent;
  font-family: ALS Hauss,Helvetica,'Helvetica New',Arial,sans-serif;
  font-weight: 500;
  text-decoration: none;
  transition: background-color 0.3s ease,color 0.3s ease,border-color 0.3s ease;
  font-size: 14px;
  line-height: 22px;
  padding: 0 20px;
  height: 36px;
  min-width: 36px;
  color: #2B2D33;
  background-color: transparent;
  padding: 0;
}

.c2:active {
  top: 1px;
}

.c2:hover {
  color: #4669C8;
}

.c3 {
  display: inline-block;
  vertical-align: top;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c3:not(:last-of-type) {
  margin-right: 8px;
}

.c3>svg {
  display: block;
}

.c5 {
  flex: 0 0 auto;
}

<div
  class="c0 className"
  id="id"
  title="title"
>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 1
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 2
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 3
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c1"
  >
    <button
      class="c2"
      type="button"
    >
      <span
        class="c3"
      />
      <span
        class="c3"
      >
        Option 4
      </span>
      <span
        class="c3"
      />
    </button>
  </div>
  <div
    class="c4"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
        type="button"
      >
        <span
          class="c3 c5"
        >
          <svg
            fill="currentColor"
            height="20"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M4 10a2 2 0 110 4 2 2 0 010-4zm16 0a2 2 0 110 4 2 2 0 010-4zm-8 0a2 2 0 110 4 2 2 0 010-4z"
            />
          </svg>
        </span>
        <span
          class="c3"
        >
          More options
        </span>
        <span
          class="c3"
        />
      </button>
    </div>
  </div>
</div>
`;
